cmake/sysbuild/partition_manager.cmake

```diff
@@ -0,0 +1,817 @@
+# - A subsys has defined a partition and CONFIG_PM_SINGLE_IMAGE is set.
+# Otherwise, return here
+#if (NOT (
+#  (IMAGE_NAME AND is_dynamic_partition_in_domain) OR
+#  (NOT IMAGE_NAME AND static_configuration) OR
+#  (NOT IMAGE_NAME AND PM_IMAGES) OR
+#  (NOT IMAGE_NAME AND PM_DOMAINS) OR
+#  (PM_SUBSYS_PREPROCESSED AND CONFIG_PM_SINGLE_IMAGE)
+#  ))
+#  return()
+#endif()
+
+# Set the dynamic partition. This is the only partition which does not
+# have a statically defined size. There is only one dynamic partition per
+# domain. For the "root domain" (ie the domain of the root image) this is
+# always "app".
+if (NOT is_dynamic_partition_in_domain)
+  set(dynamic_partition "app")  # Should this be renamed to main image name, or does it matter at all ?
+else()
+  set(dynamic_partition ${${DOMAIN}_PM_DOMAIN_DYNAMIC_PARTITION})
+  set(
+    dynamic_partition_argument
+    "--flash_primary-dynamic-partition;${dynamic_partition}"
+    )
+endif()
+
+# Image files, those should be queried from actual image.
+# Add the dynamic partition as an image partition.
+#set_property(GLOBAL PROPERTY
+#  ${dynamic_partition}_PM_HEX_FILE
+#  ${PROJECT_BINARY_DIR}/${KERNEL_HEX_NAME}
+#  )
+#
+#set_property(GLOBAL PROPERTY
+#  ${dynamic_partition}_PM_TARGET
+#  ${logical_target_for_zephyr_elf}
+#  )
+
+# Prepare the input_files, header_files, and images lists
+set(generated_path include/generated)
+# ToDo: In child image, this happens to each domain parent.
+# In parent image, it only happen to direct children, not children of children.
+# This must be adjusted into:
+# Explicitly add the main dynamic partition image
+sysbuild_get(${DEFAULT_IMAGE}_input_files IMAGE ${DEFAULT_IMAGE} VAR PM_YML_FILES CACHE)
+sysbuild_get(${DEFAULT_IMAGE}_binary_dir  IMAGE ${DEFAULT_IMAGE} VAR ZEPHYR_BINARY_DIR CACHE)
+list(APPEND prefixed_images ":${dynamic_partition}")
+list(APPEND input_files  ${${DEFAULT_IMAGE}_input_files})
+list(APPEND header_files ${${DEFAULT_IMAGE}_binary_dir}/${generated_path}/pm_config.h)
+foreach (image ${IMAGES})
+  set(domain)
+  # Special handling of `app_image` as this must be added as `:app` for historic reasons.
+  # `:app` is handled below.
+  foreach (d ${PM_DOMAINS})
+    if(${image} IN_LIST PM_${d}_IMAGES)
+      set(domain ${d})
+      break()
+    endif()
+  endforeach()
+
+  if(NOT "${DEFAULT_IMAGE}" STREQUAL "${image}")
+    sysbuild_get(${image}_input_files IMAGE ${image} VAR PM_YML_FILES CACHE)
+    sysbuild_get(${image}_binary_dir  IMAGE ${image} VAR ZEPHYR_BINARY_DIR CACHE)
+
+    list(APPEND prefixed_images ${domain}:${image})
+    list(APPEND images ${image})
+    list(APPEND header_files ${${image}_binary_dir}/${generated_path}/pm_config.h)
+    if(NOT DEFINED domain OR "${DOMAIN_APP_${domain}}" STREQUAL "${image}")
+      list(APPEND input_files  ${${image}_input_files})
+    endif()
+  endif()
+endforeach()
+
+#foreach (image ${IMAGES})
+#    # Re-configure (Re-execute all CMakeLists.txt code) when original
```

Comment:
Commented code, can this go?

Reply:
keeping for now.
Created a task for followup.